queries/query_reset.go

```diff
@@ -0,0 +1,25 @@
+package queries
+
+// QueryReset holds the state for the built up query
+type QueryReset struct {
+	q     *Query
+	undo  []undoFunc
+	saved Query
+}
+
+type undoFunc = func(*QueryReset)
+
+// Save removes the effect of a finisher
+func (q *QueryReset) Save() {
+	q.saved = *q.q
```

Comment:
This is insufficient because it does not do a deep copy. If slices/maps are modified in the query it could affect a saved query.